central/deployment/cache/cache.go

```diff
@@ -7,7 +7,11 @@ import (
 )
 
 const (
-	deletedDeploymentsRetentionPeriod = 2 * time.Minute
+	// 30 seconds for collector to send a message to sensor
+	// 30 seconds for sensor to send a message to central
+	// 5 minutes for afterglow
+	// 1 minute for a safety margin
+	deletedDeploymentsRetentionPeriod = 7 * time.Minute
```

Comment:
what are implications of changing this retention period?

Reply:
The main implication is that flows and entities that arrive to central after the afterglow period when a deployment is deleted won't be added to the database. Another is that more deployments will be kept in memory. Let's say that the largest customer has 1,000 clusters. Deleting a deployment is fairly rare, but let's say that in this case 10 deployments are deleted per minute per cluster. We are increasing this by 5 minutes so we are keeping 5*10*1,000 = 50,000 extra cluster IDs in memory. This is not very much. The memory saved by not keeping the flows and entities should be greater.

It doesn't seem like the cach of deleted deployments is used much elsewhere.